### ISSUE TITLE
[Feature Request]: Support Entra Authentication

### ISSUE THREAD

**Issue by cirdua** [RedisInsight/RedisInsight#3672]
**Problem description** 
Azure Cache for Redis allows 2 authentication methods: Access Keys and Entra Authentication.
Entra Authentication is currently not supported. and we cannot make use of the granular access it provides

**How often do you encounter this problem**
When logging into the Redis Cache using RedisInsights

**Alternatives considered** (Describe any alternative features or products you've considered).
Redis StackExchange

**Additional information**
N/A

**Comment by yuft** [comment]
I can login to Azure Redis Cache using Entra by these steps.

1. Login to Azure using CLI.
2. run `az account get-access-token --scope 'acca5fbb-b7e4-4009-81f1-37e38fd66d78/.default'`. More info at https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-azure-active-directory-for-authentication#configure-your-redis-client-to-use-microsoft-entra
3. use your Azure account's Object Id and the exposed token in the above steop to login.


Choose `Use TLS` and select TLS port number if TLS is enabled.

**Comment by RacerDelux** [comment]
Wanted to comment that this is a must. As my company adopts Azure Cache for Redis, the need to control access via Entra is high. Please consider adding this authentication method.

**Comment by ViktarStarastsenka** [comment]
this is on our Roadmap and will be available in the upcoming months.

**Comment by fardarter** [comment]
@ViktarStarastsenka That's great to hear. Would that capability propagate to Redis for VS Code by any chance?

**Comment by ViktarStarastsenka** [comment]
@fardarter , yes, the Entra ID authentication is planned for Redis for VS Code as well.
Also, you can try Azure Managed Redis, which is the most advanced Redis offering on Azure to date.

**Comment by fardarter** [comment]
@ViktarStarastsenka Thanks! I’ve been mainly looking at Azure Cache, but the prices are not too bad on Managed.

Does managed have Cuckoo support?

Our use case is very quickly checking if we’ve seen a hash before (within the last 7 days, really) and inserting if not.

**Comment by ViktarStarastsenka** [comment]
@fardarter , yes, it includes Cuckoo filter support, several performance enhancements, and a range of other updates.
An upcoming update will also introduce seamless integration with Redis Insight, such as launching Redis Insight directly from the Azure portal and easily importing Azure Managed Redis databases with just a few clicks.

Azure Managed Redis was officially released a few days ago: https://redis.io/blog/azure-managed-redis-is-ga-today/

**Comment by fardarter** [comment]
Thanks!

**Comment by RacerDelux** [comment]
> this is on our Roadmap and will be available in the upcoming months.

This is fantastic news, thank you!

**Comment by pmags** [comment]
> I can login to Azure Redis Cache using Entra by these steps.
> 
> 1. Login to Azure using CLI.
> 2. run `az account get-access-token --scope 'acca5fbb-b7e4-4009-81f1-37e38fd66d78/.default'`. More info at https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-azure-active-directory-for-authentication#configure-your-redis-client-to-use-microsoft-entra
> 3. use your Azure account's Object Id and the exposed token in the above steop to login.
> 
> Choose `Use TLS` and select TLS port number if TLS is enabled.

For anyone trying this approach please keep in mind that you need to create the link as:

rediss://<objectid>:<accesstoken>@host:port

to know the object id simply go for  Get-AzureADUser -ObjectId <user, service principal, managed identity, etc> inside az cli

**Comment by RacerDelux** [comment]
> this is on our Roadmap and will be available in the upcoming months.

Any info on this? Getting to the tail end of the year.

**Comment by ViktarStarastsenka** [comment]
@RacerDelux , no updates yet, but we’re targeting delivery in Q4.

**Comment by Ily83** [comment]
Any update @ViktarStarastsenka  ?

**Comment by ViktarStarastsenka** [comment]
@Ily83 , there is no specific date yet. However, we’re targeting delivery as soon as possible.

**Comment by RacerDelux** [comment]
> [@Ily83](https://github.com/Ily83) , there is no specific date yet. However, we’re targeting delivery as soon as possible.

With entering 2026, hopefully we should see this soon? Looks like the q4 window was sadly missed.

**Comment by Ily83** [comment]
Any news @ViktarStarastsenka ?

**Comment by Ily83** [comment]
I see https://github.com/redis/RedisInsight/pull/5465, I guess it's in the work

**Comment by pawelangelow** [comment]
Hey @Ily83, sorry I missed this thread! Yes, it’s currently in progress and will be available in the next release. I’ll keep you updated here.

**Comment by RacerDelux** [comment]
> Hey [@Ily83](https://github.com/Ily83), sorry I missed this thread! Yes, it’s currently in progress and will be available in the next release. I’ll keep you updated here.

I am very excited to hear this - my company has been in need of this for a while now!

**Comment by pawelangelow** [comment]
Apologies for the delay - it’s now on the way. We’d appreciate your feedback, so please don’t hesitate to reach out after testing.

**Comment by pawelangelow** [comment]
We've released the Azure Cache for Redis integration in [RedisInsight 3.2.0](https://github.com/redis/RedisInsight/releases/tag/3.2.0)! 🎉

Before all, we'd really appreciate any feedback on this feature:
- What's easy to use?
- What could be improved?
- Is there anything missing?
- Any ideas or suggestions?

Feel free to write here or open a new issue. 🙏 

## Setup Instructions

To use the Azure integration, you'll need to grant admin consent for the RedisInsight application in your Azure tenant:

https://github.com/redis/RedisInsight/blob/main/docs/azure-setup.md

## Next Steps
We plan to add support for access keys in some of the upcoming releases and some UI enhancements.

**Comment by DimoHG** [comment]
Closing the issue as done :)

**Comment by RacerDelux** [comment]
@pawelangelow 
Feedback from my boss:

"the entra stuff doesn't feel 'ready for prime time' to me at the moment..."

And

"setup, documentation, support [is lacking]......"

Documentation still references azure cache for redis for instance. (Even if the IDs are the same)

**Comment by pawelangelow** [comment]
@RacerDelux 

Sorry to hear this. We’re planning to expand Azure support in the next releases by adding access key authentication and improving parts of the UI.

If you could share what’s currently missing or confusing, that would help us make sure we address the right things.

**Comment by RickFrankel** [comment]
@pawelangelow I would love to be able to adjust the host details of the Azure connection.  When using private link we don't always have name resolution working from our client machines we want to manage the Redis instance from.  If we can customize the host portion of the Azure Redis connection then it allows us to connect via IP as well.

**Comment by pawelangelow** [comment]
@RickFrankel okay, so, just to clear what is expected:
1. I use the autodiscovery to add a database
2. Then  I modify this database and change the host from `something.azure.com` to `123.123.123.123`, for example

Is that the correct case?

**Comment by RacerDelux** [comment]
> If you could share what’s currently missing or confusing, that would help us make sure we address the right things.

I think it's mainly that in order for it to set up it can't be done 100% via the UI. I personally don't have an issue, but I'm more comfortable outside UI systems.

**Comment by RickFrankel** [comment]
> [@RickFrankel](https://github.com/RickFrankel) okay, so, just to clear what is expected:
> 
> 1. I use the autodiscovery to add a database
> 2. Then  I modify this database and change the host from `something.azure.com` to `123.123.123.123`, for example
> 
> Is that the correct case?

Yes.  However you cannot complete step 1 in the current flow because it won't add to the database list unless it can connect to it.
Ideally you can still add it to the list and then edit the configuration and change the host section.

<img width="647" height="407" alt="Image" src="https://github.com/user-attachments/assets/59cce8d0-0346-4b49-89ce-f86eab08f7d5" />

**Comment by pawelangelow** [comment]
> I think it's mainly that in order for it to set up it can't be done 100% via the UI. I personally don't have an issue, but I'm more comfortable outside UI systems.

Yes, that step can definitely feel a bit heavy - but it’s actually on the Azure side, not within RedisInsight.

**Why this is required:**
RedisInsight uses the PKCE (Proof Key for Code Exchange) OAuth flow to authenticate with Azure Cache for Redis. This is the recommended secure flow for public clients because it doesn’t require storing any client secrets in the application.

However, Azure’s security model requires organizations to explicitly grant consent when a third-party application requests tokens on behalf of users.

Because RedisInsight is a third-party app, an Azure AD administrator must grant consent before users can authenticate through it. Unfortunately, Azure doesn’t provide a simple one-click approval for this scenario - it requires either running CLI commands or navigating the Enterprise Applications section in the Azure Portal (which can actually be more confusing than the CLI).

**The good news:**
This is a one-time configuration per Azure tenant. Once the admin grants consent, all users in the organization can use RedisInsight with Entra ID without any additional setup.

**Comment by pawelangelow** [comment]
> Yes. However you cannot complete step 1 in the current flow because it won't add to the database list unless it can connect to it. Ideally you can still add it to the list and then edit the configuration and change the host section.

Thank you for pointing this out. We’ll work on updating the flow.
